Introducing defaulting logic (#324)
DataDog/datadog-operator: docs/configuration.md

| agent.config.dogstatsd.unixDomainSocket.enabled | Enable APM over Unix Domain Socket. See also: https://docs.datadoghq.com/developers/dogstatsd/unix_socket/ |
| agent.config.dogstatsd.unixDomainSocket.hostFilepath | Define the host APM socket filepath used when APM over Unix Domain Socket is enabled. (default value: /var/run/datadog/statsd.sock). See also: https://docs.datadoghq.com/developers/dogstatsd/unix_socket/ |
| agent.config.env | The Datadog Agent supports many environment variables. See also: https://docs.datadoghq.com/agent/docker/?tab=standard#environment-variables |

> | agent.config.healthPort | HealthPort of the agent container for internal liveness probe. Must be the same as the Liness/Readiness probes. |

Reviewer: ```suggestion
| agent.config.healthPort | HealthPort of the Agent container for internal liveness probe. Must be the same as the Liveness/Readiness probes. |
```

Or is it supposed to be Liness? I had not heard of liness probes

| agent.config.livenessProbe.successThreshold | Minimum consecutive successes for the probe to be considered successful after having failed. Defaults to 1. Must be 1 for liveness and startup. Minimum value is 1. |
| agent.config.livenessProbe.tcpSocket.host | Optional: Host name to connect to, defaults to the pod IP. |
| agent.config.livenessProbe.tcpSocket.port | Number or name of the port to access on the container. Number must be in the range 1 to 65535. Name must be an IANA_SVC_NAME. |
| agent.config.livenessProbe.timeoutSeconds | Number of seconds after which the probe times out. Defaults to 1 second. Minimum value is 1. More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle#container-probes |
| agent.config.logLevel | Set logging verbosity, valid log levels are: trace, debug, info, warn, error, critical, and off |
| agent.config.podAnnotationsAsTags | Provide a mapping of Kubernetes Annotations to Datadog Tags. <KUBERNETES_ANNOTATIONS>: <DATADOG_TAG_KEY> |
| agent.config.podLabelsAsTags | Provide a mapping of Kubernetes Labels to Datadog Tags. <KUBERNETES_LABEL>: <DATADOG_TAG_KEY> |

> | agent.config.readinessProbe.exec.command | Command is the command line to execute inside the container, the working directory for the command  is root ('/') in the container's filesystem. The command is simply exec'd, it is not run inside a shell, so traditional shell instructions ('|', etc) won't work. To use a shell, you need to explicitly call out to that shell. Exit status of 0 is treated as live/healthy and non-zero is unhealthy. |

Reviewer: ```suggestion
| agent.config.readinessProbe.exec.command | Command is the command line to execute inside the container. The working directory for the command is root ('/') in the container's filesystem. The command is simply executed; it is not run inside a shell, so traditional shell instructions ('|', etc.) do not work. To use a shell, you need to explicitly call out to that shell. Exit status of 0 is treated as live/healthy and non-zero is unhealthy. |
```

> | agent.config.readinessProbe.failureThreshold | Minimum consecutive failures for the probe to be considered failed after having succeeded. Defaults to 3. Minimum value is 1. |

Reviewer: ```suggestion
| agent.config.readinessProbe.failureThreshold | Minimum consecutive failures for the probe to be considered failed after having succeeded. Defaults to `3`. Minimum value is `1`. |
```

> | agent.config.readinessProbe.httpGet.host | Host name to connect to, defaults to the pod IP. You probably want to set "Host" in httpHeaders instead. |

Reviewer: ```suggestion
| agent.config.readinessProbe.httpGet.host | Host name to connect to, defaults to the pod IP. You may want to set "Host" in `httpHeaders` (see next parameter) instead. |
```

| agent.config.readinessProbe.httpGet.httpHeaders | Custom headers to set in the request. HTTP allows repeated headers. |
| agent.config.readinessProbe.httpGet.path | Path to access on the HTTP server. |

> | agent.config.readinessProbe.httpGet.port | Name or number of the port to access on the container. Number must be in the range 1 to 65535. Name must be an IANA_SVC_NAME. |

Reviewer: ```suggestion
| agent.config.readinessProbe.httpGet.port | Name or number of the port to access on the container. Number must be in the range `1` to `65535`. Name must be an `IANA_SVC_NAME`. |
```

| agent.config.readinessProbe.httpGet.scheme | Scheme to use for connecting to the host. Defaults to HTTP. |
| agent.config.readinessProbe.initialDelaySeconds | Number of seconds after the container has started before liveness probes are initiated. More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle#container-probes |

> | agent.config.readinessProbe.periodSeconds | How often (in seconds) to perform the probe. Default to 10 seconds. Minimum value is 1. |

Reviewer: ```suggestion
| agent.config.readinessProbe.periodSeconds | How often (in seconds) to perform the probe. Default to `10` seconds. Minimum value is `1`. |
```

> | agent.config.readinessProbe.successThreshold | Minimum consecutive successes for the probe to be considered successful after having failed. Defaults to 1. Must be 1 for liveness and startup. Minimum value is 1. |

Reviewer: ```suggestion
| agent.config.readinessProbe.successThreshold | Minimum consecutive successes for the probe to be considered successful after having failed. Defaults to `1`. Must be `1` for liveness and startup. Minimum value is `1`. |
```

| agent.config.readinessProbe.tcpSocket.host | Optional: Host name to connect to, defaults to the pod IP. |

> | agent.config.readinessProbe.tcpSocket.port | Number or name of the port to access on the container. Number must be in the range 1 to 65535. Name must be an IANA_SVC_NAME. |

Reviewer: ```suggestion
| agent.config.readinessProbe.tcpSocket.port | Number or name of the port to access on the container. Number must be in the range `1` to `65535`. Name must be an `IANA_SVC_NAME`. |
```

> | agent.config.readinessProbe.timeoutSeconds | Number of seconds after which the probe times out. Defaults to 1 second. Minimum value is 1. More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle#container-probes |

Reviewer: ```suggestion
| agent.config.readinessProbe.timeoutSeconds | Number of seconds after which the probe times out. Defaults to `1` second. Minimum value is `1`. More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle#container-probes |
```

| clusterAgent.config.externalMetrics.port | If specified configures the metricsProvider external metrics service port. |
| clusterAgent.config.externalMetrics.useDatadogMetrics | Enable usage of DatadogMetrics CRD (allow to scale on arbitrary queries). |
| clusterAgent.config.externalMetrics.wpaController | Enable informer and controller of the watermark pod autoscaler. NOTE: The WatermarkPodAutoscaler controller needs to be installed. See also: https://github.com/DataDog/watermarkpodautoscaler. |

> | clusterAgent.config.healthPort | HealthPort of the agent container for internal liveness probe. Must be the same as the Liness/Readiness probes. |

Reviewer: ```suggestion
| clusterAgent.config.healthPort | HealthPort of the Agent container for internal liveness probe. Must be the same as the Liveness/Readiness probes. |
```

Should it be "liveness" instead of liness?

| clusterChecksRunner.config.args | Args allows the specification of extra args to `Command` parameter |
| clusterChecksRunner.config.command | Command allows the specification of custom entrypoint for Cluster Checks Runner container |
| clusterChecksRunner.config.env | The Datadog Agent supports many environment variables. See also: https://docs.datadoghq.com/agent/docker/?tab=standard#environment-variables |

> | clusterChecksRunner.config.healthPort | HealthPort of the agent container for internal liveness probe. Must be the same as the Liness/Readiness probes. |

Reviewer: ```suggestion
| clusterChecksRunner.config.healthPort | HealthPort of the Agent container for internal liveness probe. Must be the same as the Liveness/Readiness probes. |
```
again i'm not 100% sure this is supposed to be liveness vs. liness

> | clusterChecksRunner.config.livenessProbe.exec.command | Command is the command line to execute inside the container, the working directory for the command  is root ('/') in the container's filesystem. The command is simply exec'd, it is not run inside a shell, so traditional shell instructions ('|', etc) won't work. To use a shell, you need to explicitly call out to that shell. Exit status of 0 is treated as live/healthy and non-zero is unhealthy. |

Reviewer: ```suggestion
| clusterChecksRunner.config.livenessProbe.exec.command | Command is the command line to execute inside the container. The working directory for the command is root ('/') in the container's filesystem. The command is simply exec'd; it is not run inside a shell, so traditional shell instructions ('|', etc) do not work. To use a shell, you need to explicitly call out to that shell. Exit status of 0 is treated as live/healthy and non-zero is unhealthy. |
```

> | clusterChecksRunner.config.livenessProbe.failureThreshold | Minimum consecutive failures for the probe to be considered failed after having succeeded. Defaults to 3. Minimum value is 1. |

Reviewer: ```suggestion
| clusterChecksRunner.config.livenessProbe.failureThreshold | Minimum consecutive failures for the probe to be considered failed after having succeeded. Defaults to `3`. Minimum value is `1`. |
```

> | clusterChecksRunner.config.livenessProbe.httpGet.host | Host name to connect to, defaults to the pod IP. You probably want to set "Host" in httpHeaders instead. |

Reviewer: ```suggestion
| clusterChecksRunner.config.livenessProbe.httpGet.host | Host name to connect to, defaults to the pod IP. You may want to set "Host" in `httpHeaders` instead. |
```

| clusterChecksRunner.config.livenessProbe.httpGet.httpHeaders | Custom headers to set in the request. HTTP allows repeated headers. |
| clusterChecksRunner.config.livenessProbe.httpGet.path | Path to access on the HTTP server. |

> | clusterChecksRunner.config.livenessProbe.httpGet.port | Name or number of the port to access on the container. Number must be in the range 1 to 65535. Name must be an IANA_SVC_NAME. |

Reviewer: ```suggestion
| clusterChecksRunner.config.livenessProbe.httpGet.port | Name or number of the port to access on the container. Number must be in the range `1` to `65535`. Name must be an `IANA_SVC_NAME`. |
```

| clusterChecksRunner.config.livenessProbe.httpGet.scheme | Scheme to use for connecting to the host. Defaults to HTTP. |
| clusterChecksRunner.config.livenessProbe.initialDelaySeconds | Number of seconds after the container has started before liveness probes are initiated. More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle#container-probes |

> | clusterChecksRunner.config.livenessProbe.periodSeconds | How often (in seconds) to perform the probe. Default to 10 seconds. Minimum value is 1. |

Reviewer: ```suggestion
| clusterChecksRunner.config.livenessProbe.periodSeconds | How often (in seconds) to perform the probe. Defaults to `10` seconds. Minimum value is `1`. |
```

> | clusterChecksRunner.config.livenessProbe.successThreshold | Minimum consecutive successes for the probe to be considered successful after having failed. Defaults to 1. Must be 1 for liveness and startup. Minimum value is 1. |

Reviewer: ```suggestion
| clusterChecksRunner.config.livenessProbe.successThreshold | Minimum consecutive successes for the probe to be considered successful after having failed. Defaults to `1`. Must be `1` for liveness and startup. Minimum value is `1`. |
```

| clusterChecksRunner.config.livenessProbe.tcpSocket.host | Optional: Host name to connect to, defaults to the pod IP. |

> | clusterChecksRunner.config.livenessProbe.tcpSocket.port | Number or name of the port to access on the container. Number must be in the range 1 to 65535. Name must be an IANA_SVC_NAME. |

Reviewer: ```suggestion
| clusterChecksRunner.config.livenessProbe.tcpSocket.port | Number or name of the port to access on the container. Number must be in the range `1` to `65535`. Name must be an `IANA_SVC_NAME`. |
```

> | clusterChecksRunner.config.livenessProbe.timeoutSeconds | Number of seconds after which the probe times out. Defaults to 1 second. Minimum value is 1. More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle#container-probes |

Reviewer: ```suggestion
| clusterChecksRunner.config.livenessProbe.timeoutSeconds | Number of seconds after which the probe times out. Defaults to `1` second. Minimum value is `1`. More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle#container-probes |
```

| clusterChecksRunner.config.logLevel | Set logging verbosity, valid log levels are: trace, debug, info, warn, error, critical, and off |

> | clusterChecksRunner.config.readinessProbe.exec.command | Command is the command line to execute inside the container, the working directory for the command  is root ('/') in the container's filesystem. The command is simply exec'd, it is not run inside a shell, so traditional shell instructions ('|', etc) won't work. To use a shell, you need to explicitly call out to that shell. Exit status of 0 is treated as live/healthy and non-zero is unhealthy. |

Reviewer: ```suggestion
| clusterChecksRunner.config.readinessProbe.exec.command | Command is the command line to execute inside the container. The working directory for the command is root ('/') in the container's filesystem. The command is simply exec'd; it is not run inside a shell, so traditional shell instructions ('|', etc.) do not work. To use a shell, you need to explicitly call out to that shell. Exit status of 0 is treated as live/healthy and non-zero is unhealthy. |
```

> | clusterChecksRunner.config.readinessProbe.failureThreshold | Minimum consecutive failures for the probe to be considered failed after having succeeded. Defaults to 3. Minimum value is 1. |

Reviewer: ```suggestion
| clusterChecksRunner.config.readinessProbe.failureThreshold | Minimum consecutive failures for the probe to be considered failed after having succeeded. Defaults to `3`. Minimum value is `1`. |
```

> | clusterChecksRunner.config.readinessProbe.httpGet.host | Host name to connect to, defaults to the pod IP. You probably want to set "Host" in httpHeaders instead. |

Reviewer: ```suggestion
| clusterChecksRunner.config.readinessProbe.httpGet.host | Host name to connect to, defaults to the pod IP. You may want to set "Host" in `httpHeaders` instead. |
```

| clusterChecksRunner.config.readinessProbe.httpGet.httpHeaders | Custom headers to set in the request. HTTP allows repeated headers. |
| clusterChecksRunner.config.readinessProbe.httpGet.path | Path to access on the HTTP server. |

> | clusterChecksRunner.config.readinessProbe.httpGet.port | Name or number of the port to access on the container. Number must be in the range 1 to 65535. Name must be an IANA_SVC_NAME. |

Reviewer: ```suggestion
| clusterChecksRunner.config.readinessProbe.httpGet.port | Name or number of the port to access on the container. Number must be in the range `1` to `65535`. Name must be an `IANA_SVC_NAME`. |
```

| clusterChecksRunner.config.readinessProbe.httpGet.scheme | Scheme to use for connecting to the host. Defaults to HTTP. |
| clusterChecksRunner.config.readinessProbe.initialDelaySeconds | Number of seconds after the container has started before liveness probes are initiated. More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle#container-probes |

> | clusterChecksRunner.config.readinessProbe.periodSeconds | How often (in seconds) to perform the probe. Default to 10 seconds. Minimum value is 1. |

Reviewer: ```suggestion
| clusterChecksRunner.config.readinessProbe.periodSeconds | How often (in seconds) to perform the probe. Default to `10` seconds. Minimum value is `1`. |
```

> | clusterChecksRunner.config.readinessProbe.successThreshold | Minimum consecutive successes for the probe to be considered successful after having failed. Defaults to 1. Must be 1 for liveness and startup. Minimum value is 1. |

Reviewer: ```suggestion
| clusterChecksRunner.config.readinessProbe.successThreshold | Minimum consecutive successes for the probe to be considered successful after having failed. Defaults to `1`. Must be `1` for liveness and startup. Minimum value is `1`. |
```

| clusterChecksRunner.config.readinessProbe.tcpSocket.host | Optional: Host name to connect to, defaults to the pod IP. |

> | clusterChecksRunner.config.readinessProbe.tcpSocket.port | Number or name of the port to access on the container. Number must be in the range 1 to 65535. Name must be an IANA_SVC_NAME. |

Reviewer: ```suggestion
| clusterChecksRunner.config.readinessProbe.tcpSocket.port | Number or name of the port to access on the container. Number must be in the range `1` to `65535`. Name must be an `IANA_SVC_NAME`. |
```

> | clusterChecksRunner.config.readinessProbe.timeoutSeconds | Number of seconds after which the probe times out. Defaults to 1 second. Minimum value is 1. More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle#container-probes |

Reviewer: ```suggestion
| clusterChecksRunner.config.readinessProbe.timeoutSeconds | Number of seconds after which the probe times out. Defaults to `1` second. Minimum value is `1`. More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle#container-probes |
```
